includes/enrolment/class-sensei-course-enrolment.php

```diff
@@ -444,4 +454,98 @@ public function reset_course_enrolment_salt() {
 
 		return $new_salt;
 	}
+
+	/**
+	 * Remove learner from the course, overriding the providers rule.
+	 *
+	 * @param int $user_id User ID.
+	 *
+	 * @return boolean Success flag.
+	 */
+	public function remove_learner( $user_id ) {
+		$removed_learners = $this->get_removed_learners();
+
+		if ( isset( $removed_learners[ $user_id ] ) ) {
+			return false;
+		}
+
+		$removed_learners[ $user_id ] = [ 'date' => time() ];
+
+		$this->save_enrolment( $user_id, false );
+
+		return $this->update_removed_learners( $removed_learners );
+	}
+
+	/**
+	 * Restore removed learner enrolment, giving the control back to the providers.
+	 *
+	 * @param int $user_id User ID.
+	 *
+	 * @return boolean Success flag.
+	 */
+	public function restore_removed_learner( $user_id ) {
```

Comment:
What about calling it `restore_learner`? Restore already implies that the learner was removed so it's a bit redundant.

Reply:
Indeed 899ec91e9d7b0aabb96a2ed826474dbaaa054a75